cmd/argoexec/commands/wait.go

```diff
@@ -61,10 +59,24 @@ func waitContainer(ctx context.Context) error {
 	if err != nil {
 		wfExecutor.AddError(err)
 	}
-	// Annotating pod with output
-	err = wfExecutor.AnnotateOutputs(ctx, logArt)
-	if err != nil {
-		wfExecutor.AddError(err)
+
+	containerNames := wfExecutor.Template.GetMainContainerNames()
```

Comment:
could this all go inside SaveLogs?

Reply:
i moved the iteration over the containerNames into `SaveLogs`, and from there (inside the `WorkflowExecutor`) i made the calls to `we.AddError(err)` when needed.